code/modules/projectiles/projectile/magic.dm

```diff
@@ -451,6 +451,142 @@
+			return BULLET_ACT_BLOCK
+		SEND_SIGNAL(M, COMSIG_ADD_MOOD_EVENT, src, /datum/mood_event/sapped)
+
+/obj/item/projectile/magic/necropotence
+	name = "bolt of necropotence"
+	icon_state = "necropotence"
+
+/obj/item/projectile/magic/necropotence/on_hit(target)
+	. = ..()
+	if(isliving(target))
+		var/mob/living/L = target
+		if(L.anti_magic_check() || !L.mind || !L.mind.hasSoul)
+			L.visible_message("<span class='warning'>[src] vanishes on contact with [target]!</span>")
+			return BULLET_ACT_BLOCK
+		to_chat(L, "<span class='danger'>Your body feels drained and there is a burning pain in your chest.</span>")
+		L.maxHealth -= 20
+		L.health = min(L.health, L.maxHealth)
+		if(L.maxHealth <= 0)
+			to_chat(L, "<span class='userdanger'>Your weakened soul is completely consumed by the [src]!</span>")
+			L.mind.hasSoul = FALSE
+		for(var/obj/effect/proc_holder/spell/spell in L.mind.spell_list)
+			spell.charge_counter = spell.charge_max
+			spell.recharging = FALSE
+			spell.update_icon()
+
+/obj/item/projectile/magic/wipe
+	name = "bolt of possession"
+	icon_state = "wipe"
+
+/obj/item/projectile/magic/wipe/on_hit(target)
+	. = ..()
+	if(iscarbon(target))
+		var/mob/living/carbon/M = target
+		if(M.anti_magic_check())
+			M.visible_message("<span class='warning'>[src] vanishes on contact with [target]!</span>")
+			return BULLET_ACT_BLOCK
+		for(var/x in M.get_traumas())//checks to see if the victim is already going through possession
+			if(istype(x, /datum/brain_trauma/special/imaginary_friend/trapped_owner))
+				M.visible_message("<span class='warning'>[src] vanishes on contact with [target]!</span>")
+				return BULLET_ACT_BLOCK
+		to_chat(M, "<span class='warning'>Your mind has been opened to possession!</span>")
+		possession_test(M)
+		return BULLET_ACT_BLOCK
+
+/obj/item/projectile/magic/wipe/proc/possession_test(var/mob/living/carbon/M)
+	var/datum/brain_trauma/special/imaginary_friend/trapped_owner/trauma = M.gain_trauma(/datum/brain_trauma/special/imaginary_friend/trapped_owner)
+	var/poll_message = "Do you want to play as [M.real_name]?"
+	if(M.mind && M.mind.assigned_role)
+		poll_message = "[poll_message] Job:[M.mind.assigned_role]."
+	if(M.mind && M.mind.special_role)
+		poll_message = "[poll_message] Status:[M.mind.special_role]."
+	else if(M.mind)
+		var/datum/antagonist/A = M.mind.has_antag_datum(/datum/antagonist/)
+		if(A)
+			poll_message = "[poll_message] Status:[A.name]."
+	var/list/mob/dead/observer/candidates = pollCandidatesForMob(poll_message, ROLE_PAI, null, FALSE, 100, M)
+	if(M.stat == DEAD)//boo.
+		return
+	if(LAZYLEN(candidates))
+		var/mob/dead/observer/C = pick(candidates)
+		to_chat(M, "You have been noticed by a ghost, and it has possessed you!")
+		var/oldkey = M.key
+		M.ghostize(0)
+		M.key = C.key
+		trauma.friend.key = oldkey
+		trauma.friend.reset_perspective(null)
+		trauma.friend.Show()
+		trauma.friend_initialized = TRUE
```

Comment:
isn't there a transfer mind proc that's safer than just forcing ckey changes?

I'm pretty sure doing that kicks people from the game lol.

Reply:
it's what offer to ghosts does, and what sentience balloons do, do you know where to check for this proc?

Reply:
nvm u ghost first